### ISSUE TITLE
Tracking Issue for doc --check

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/cargo#10025]
### Summary

The `--check` option has been implemented in https://github.com/rust-lang/rust/pull/78984. It'd be nice to be able to use it through cargo.

### Future Extensions

When the `--check` is stabilized on rustdoc, it'd be nice to run it by default on `cargo check`.

### About tracking issues

Tracking issues are used to record the overall progress of implementation.
They are also used as hubs connecting to other relevant issues, e.g., bugs or open design questions.
A tracking issue is however *not* meant for large scale discussion, questions, or bug reports about a feature.
Instead, open a dedicated issue for the specific matter and add the relevant feature gate label.

**Comment by Manishearth** [comment]
cc @rust-lang/rustdoc

**Comment by Nemo157** [comment]
There would be a little inconsistency between `cargo test --doc` and `cargo check --doc` if the latter checks the docs but not the doc-examples.

**Comment by gilescope** [comment]
nice - this is not running the tests I assume? Would love `cargo check --all` to include `doc --check`.

**Comment by jyn514** [comment]
> nice - this is not running the tests I assume? Would love cargo check --all to include doc --check.

No, this doesn't currently check doctests. (I assume that's what you mean - *running* the tests in a command called `check` seems very confusing).

There's a lot of uncertainty around here right now - here's the most recent discussion: https://github.com/rust-lang/rust/pull/73314#issuecomment-771638112
That suggests having `check --doc` and `check --doctests` be different, but still both be included in `--all-targets`. I like that idea, I think it's confusing to combine doctests and rustdoc lints.

**Comment by jyn514** [comment]
https://github.com/rust-lang/cargo/pull/10026#issuecomment-962003876
> There would be a little inconsistency between cargo test --doc and cargo check --doc if the latter checks the docs but not the doc-examples.

@Nemo157 I agree this is confusing, but I don't know if it's *more* confusing that combining the two; I think it's better to lean on the side of giving people more options.
